### PR TITLE
boards: add initial support for ST b-l475e-iot01a

### PR DESCRIPTION
**Pull request by aabadie**
Board datasheet is [here](http://www.st.com/en/evaluation-tools/b-l475e-iot01a.html), detailed pdf is [here](http://www.st.com/content/ccc/resource/technical/document/user_manual/group0/b1/b8/7a/f2/f7/8d/4b/6b/DM00347848/files/DM00347848.pdf/jcr:content/translations/en.DM00347848.pdf).

This board is very interesting for IoT:
* it is low power: uses a STM32L475VG mcu
* it embeds lots of radio devices: WiFi, BLE, NFC and LoRa
* it embeds lots of sensors: microphone (MP34DT01), temperature/humidy(HTS221), magnetometer (LIS3MDL), accelerometer (LSM6DSL), pressure (LPS22HB), TOF ranging (VL53L0X)
* it's arduino pinout compatible (not very important though)

I'll create an issue to keep track of the remaining things to do to fully support it.

Thanks to @kYc0o who pointed me to it :)

### CODE REVIEW

**Comment by gebart** [comment]
That's a lot of connectivity options on one board.

**Comment by aabadie** [comment]
I received some gifts today with 2 of these boards in  :)

Just tested it and this PR seems to work: I could play with the shell and on-board LEDs. I need to go further in testing and also rebase.

**Comment by aabadie** [comment]
The initial port was not that bad. I tested configured uart, timer, rtc, rtt with success.

Maybe @astralien3000 you can test the configured PWMs ?

File: boards/b-l475e-iot01a/Makefile.include [hunk 1]
```diff
@@ -0,0 +1,13 @@
+## the cpu to build for
```
Comment (haukepetersen) [inline boards/b-l475e-iot01a/Makefile.include]: extra `#` char?!

File: boards/b-l475e-iot01a/board.c [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+#include "board.h"
+#include "periph/gpio.h"
+
+void board_init(void)
+{
+    /* initialize the CPU */
+    cpu_init();
+
+    /* initialize the boards LED */
+#ifdef AUTO_INIT_LED0
```
Comment (haukepetersen) [inline boards/b-l475e-iot01a/board.c]: just to make sure: is this also the case on this board or is this just a copy/paste error from the nucleo code?

Comment (aabadie) [reply boards/b-l475e-iot01a/board.c]: It's also the case on this board => LED_0 connected to SPI_0 clock

Comment (aabadie) [reply boards/b-l475e-iot01a/board.c]: > is this just a copy/paste error from the nucleo code?

hopefully all my code is not only copy paste

**Comment by aabadie** [comment]
> the discussion from #8058 applies here as well -> 90% of the lines in this PR are duplicates of existing code

I know, that's why I'm very much in favour of the proposed solutions there (whichever is chosen). See [my comment](https://github.com/RIOT-OS/RIOT/pull/8058#issuecomment-345176946) talking about the introduction of a `boards/common/stm` directory that also goes in that direction.

The board here and others from ST are sharing a lot of things (Arduino compatible pinout, ST-Link, etc)

**Comment by aabadie** [comment]
I would also like to have #7799 merged soon and rebase on it after

File: boards/b-l475e-iot01a/Makefile.include (hunk 1 above)
Comment (aabadie) [reply boards/b-l475e-iot01a/Makefile.include]: fixed

**Comment by kYc0o** [comment]
> The board here and others from ST are sharing a lot of things (Arduino compatible pinout, ST-Link, etc)

Take a look at #7686 to adapt the debugger.

**Comment by astralien3000** [comment]
Saul SENSE_BTN does not seem to work.

**Comment by aabadie** [comment]
> it's C13 instead of B2

Thanks for having a look @astralien3000 ! Will fix

**Comment by aabadie** [comment]
@astralien3000, I fixed the wrong pin definitions and removed the 2 PWM that doesn't work as it seems to be a more general driver issue (not being able to change the timer width).

**Comment by aabadie** [comment]
@kYc0o, I addressed your comment regarding the debugger configuration. Still works !

**Comment by aabadie** [comment]
@kYc0o, now that #7799 is merged, I rebased this PR and remove the ld script. Just tested and it still works.

**Comment by aabadie** [comment]
@haukepetersen, @kYc0o, I think we are good with this one. I tested it again locally and it's still working.

File: boards/b-l475e-iot01a/Makefile.features [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# Put defined MCU peripherals here (in alphabetical order)
+FEATURES_PROVIDED += periph_gpio
+FEATURES_PROVIDED += periph_pwm
+FEATURES_PROVIDED += periph_spi
+FEATURES_PROVIDED += periph_timer
+FEATURES_PROVIDED += periph_uart
+
+# Load extra provided features
```
[the remark this answers is not in the document]
Comment (aabadie) [reply boards/b-l475e-iot01a/Makefile.features]: Done

**Comment by fernandoavita** [comment]
The LED1 macros are not working due to a typo in the board.h file. They are defined with LED0_MASK instead of LED1_MASK.

Thanks adding support to this board! I would like to learn RIOT and I'm with this board in hands.

**Comment by aabadie** [comment]
> The LED1 macros are not working due to a typo in the board.h file

Thanks for reporting @fernandoavita. I just pushed the fixed configuration. Can you test again ?

**Comment by fernandoavita** [comment]
>  I just pushed the fixed configuration. Can you test again ?

Sure! Now they are working. Thanks @aabadie!

**Comment by aabadie** [comment]
@kaspar030 ? (now that your github notifications are fixed :) )

**Pull request merged**